Resources/Storage/OccupancyPlugins/WLCGAccountingJson.py

```diff
@@ -0,0 +1,121 @@
+
+from DIRAC import gLogger, gConfig
+from DIRAC import S_OK, S_ERROR
+
+
+class WLCGAccountingJson(object):
+  """ .. class:: WLCGAccountingJson
+
+  Occupancy plugin to return the space information given by WLCG Accouting Json
+  """
+  def __init__(self, se):
+    self.se = se
+    self.log = se.log.getSubLogger('WLCGAccountingJson')
+    self.name = self.se.name
+
+  def getOccupancy(self, **kwargs):
+    """ Returns the space information given by LCG Accouting Json
+
+        :returns: S_OK with dict (keys: SpaceReservation, Total, Free)
+    """
+    occupancyLFN = kwargs['occupancyLFN']
+
+    if not occupancyLFN:
+      return S_ERROR("Failed to get occupancyLFN")
+
+    tmpDirName = tempfile.mkdtemp()
+    filePath = os.path.join(tmpDirName, os.path.basename(occupancyLFN))
+
+    for storage in self.se.storages:
+      try:
+        ctx = gfal2.creat_context()
+        params = ctx.transfer_parameters()
+        params.overwrite = True
+        res = storage.updateURL(occupancyLFN)
+        if not res['OK']:
+          return res
+        occupancyURL = res['Value']
+        ctx.filecopy(params, occupancyURL, 'file://' + filePath)
+
+      except gfal2.GError as e:
+        detailMsg = "Failed to copy file %s to destination url %s: [%d] %s" % (
+            occupancyURL, filePath, e.code, e.message)
+        self.log.debug("Exception while copying", detailMsg)
+        continue
+
+      else:
+        break
+
+    if not os.path.isfile(filePath):
+      return S_ERROR('No WLCGAccountingJson file of %s is downloaded.' % (self.name))
+
+    with open(filePath, 'r') as path:
+      occupancyDict = json.load(path)
+
+    # delete temp dir
+    shutil.rmtree(tmpDirName)
+
+    if 'storageservice' not in occupancyDict:
+      return S_ERROR('Could not find storageservice component in %s at %s' % (occupancyLFN, self.name))
+    storageService = occupancyDict['storageservice']
+
+    if 'storageshares' not in storageService:
+      return S_ERROR('Could not find storageshares component in %s at %s' % (occupancyLFN, self.name))
+    storageShares = occupancyDict['storageservice']['storageshares']
+
+    # get storageReservation
+    spaceReservation = self.se.options.get('SpaceReservation')
+    if not spaceReservation:
+      self.log.debug(
+          'Get SpaceToken in storage parameters instead of SpaceReservation because it is not defined in CS')
+      for storage in self.se.storages:
+        SEparams = storage.getParameters()
+        if not SEparams:
+          self.log.debug('Could not get storage parameters at %s' % (self.name))
+          continue
+        if 'SpaceToken' in SEparams:
+          spaceReservation = SEparams['SpaceToken']
+          break
+        else:
+          self.log.debug('Could not find SpaceToken key in storage parameters at %s' % (self.name))
+          continue
+
+    # get storageshares in WLCGAccountingJson file
+    storageSharesSR = None
+    if spaceReservation:
+      for key in storageShares:
```

Comment:
please avoid name like `key`, because they are not very descriptive :) maybe use `share` or something like this, which describes better the object ? :)